File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int best = 0;
        int start = 0;
        vector<int> seen(256,-1);
        for (int i=0; i<(int)s.size(); i++){
            char ch = s[i];
            if (seen[ch] != -1)
                start = max(start,seen[ch]+1);
            int len = i-start+1;
            best = max(best,len);
            seen[ch] = i;
        }
        return best;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen  = [-1] * 256 
        start = 0
        best  = 0
        for i,ch in enumerate(s):
            if seen[ord(ch)] != -1:
                start = max(start,seen[ord(ch)]+1)
            best = max(best,i-start+1)
            seen[ord(ch)] = i
        return best
